src/qcodes/instrument_drivers/cryomagnetics/Cryomagnetics4G_visa.py

```diff
@@ -0,0 +1,287 @@
+import re
+import time
+from dataclasses import dataclass
+
+from qcodes import VisaInstrument
+from qcodes.utils.validators import Enum, Numbers
+
+
+@dataclass
+class CryomagneticsOperatingState:
+    ramping: bool = False
+    holding : bool = False
+    standby: bool = False
+    quench_condition_present: bool = False
+    power_module_failure:  bool = False
+
+    def _can_start_ramping(self) -> bool:
+        required_checks = [
+            "ramping",
+            "quench_condition_present",
+            "power_module_failure",
+        ]
+        return all(not getattr(self, field) for field in required_checks)
```

Comment:
Its a bit surprising to see a private method that is only used outside the class. Should this perhaps be public?